staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -26,10 +26,31 @@ import (
 	"strings"
 	"unicode"
 
+	jsonutil "k8s.io/apimachinery/pkg/util/json"
+
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/yaml"
 )
 
+// Unmarshal unmarshals the given data
+// If v is a *map[string]interface{}, numbers are converted to int64 or float64
+func Unmarshal(data []byte, v interface{}) error {
+	if err := yaml.Unmarshal(data, v, func(d *json.Decoder) *json.Decoder {
+		// preserve int/float distinctions
+		d.UseNumber()
```

Comment:
we should either:
* make this consistent with util/json and handle all three types of interface objects, or
* only set d.UseNumber() if v is a `*map[string]interface{}` and we're going to clean up the json.Number items

Reply:
Thanks for spotting! At first I thought to add only `*map[string]interface{}` until we have a use case for the others cases, however thinking about it I kind of like `yaml.Unmarshal` and `json.Unmarshal` to be consistent.

What do you think about exposing all those `convert.*Numbers` methods?

Reply:
yeah, I don't see a reason not to be consistent

Reply:
Cool, I have updated my commit and changed a bit the test as well.